frontend/lib/src/components/widgets/Slider/Slider.tsx

```diff
@@ -242,6 +232,54 @@ class Slider extends React.PureComponent<Props, State> {
     }
   }
 
+  private thumbValueAlignment(): void {
+    const sliderDiv = this.sliderRef.current
+    const thumb1Div = this.thumbRef[0]?.current
+    const thumb2Div = this.thumbRef[1]?.current
+    const thumb1ValueDiv = this.thumbValueRef[0]?.current
+    const thumb2ValueDiv = this.thumbValueRef[1]?.current
+    // Minimum gap between thumb values (in px)
+    const labelGap = 16
+
+    // Handles label alignment over each thumb
+    this.alignValueOnThumb(sliderDiv, thumb1Div, thumb1ValueDiv)
+    this.alignValueOnThumb(sliderDiv, thumb2Div, thumb2ValueDiv)
+
+    // Checks & handles label spacing when two thumb values & they overlap
+    if (
+      sliderDiv &&
+      thumb1Div &&
+      thumb2Div &&
+      thumb1ValueDiv &&
+      thumb2ValueDiv
+    ) {
+      const slider = sliderDiv.getBoundingClientRect()
+      const thumb1 = thumb1Div.getBoundingClientRect()
+      const thumb2 = thumb2Div.getBoundingClientRect()
+      const thumb1Value = thumb1ValueDiv.getBoundingClientRect()
+      const thumb2Value = thumb2ValueDiv.getBoundingClientRect()
+
+      // Check if thumb values are overlapping or too close together
+      if (thumb1Value.right + labelGap > thumb2Value.left) {
+        // Check whether to shift 1st thumb value left or 2nd thumb value right
+        const moveLeft =
+          thumb2Value.left - labelGap - thumb1Value.width > slider.left
+        const moveRight =
```

Comment:
nit: Is there a situation where both `moveLeft` & `moveRight` are false at the same time? Otherwise, there could be also just a single check e.g. `moveLeft`, and do the move to the right in the else clause.

Reply:
Good point, since we've already separately validated overlap. Will update 👍🏼